src/wp-includes/class-wp-query.php

```diff
@@ -4714,11 +4714,20 @@ public function generate_postdata( $post ) {
 
 		$authordata = get_userdata( $post->post_author );
 
-		$currentday   = mysql2date( 'd.m.y', $post->post_date, false );
-		$currentmonth = mysql2date( 'm', $post->post_date, false );
-		$numpages     = 1;
-		$multipage    = 0;
-		$page         = $this->get( 'page' );
+		$post_date = $post->post_date;
+		if ( ! empty( $post_date ) && '0000-00-00 00:00:00' !== $post_date ) {
+			list( $year, $month, $day ) = explode( '-', substr( $post_date, 0, 10 ) );
+
+			$currentday   = sprintf( '%s.%s.%s', $day, $month, substr( $year, -2 ) );
+			$currentmonth = $month;
+		} else {
+			$currentday   = false;
```

Comment:
I would prefer if values were instantiated to `false` defaults above the `if` branch. The `else` and logic/nesting level it adds is unnecessary in this case.